include/boost/geometry/algorithms/HausdorffDistance.hpp

```diff
@@ -0,0 +1,144 @@
+/*  
+        Boost (Gsoc '18)
+        SimmilarityBetweenGeometries
+        Author : Yaghyavardhan Sing Khangarot
+        Institute: IIIT Hyderabad
+*/
+
+#include <algorithm> 
+
+#ifdef BOOST_GEOMETRY_DEBUG_FRECHET_DISTANCE
+#include <iostream>
+#endif
+
+#include <iterator>
+#include <utility>
+#include <vector>
+#include <limits>
+
+#include <boost/geometry/geometry.hpp>
+#include <boost/geometry/geometries/linestring.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/range.hpp>
+#include <boost/mpl/assert.hpp>
+#include <boost/geometry/core/tag.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace hausdorff_distance
+{
+
+struct linestring_linestring
+{
+    template <typename Linestring1, typename Linestring2, typename Strategy>
+    static inline typename distance_result
+        <
+            typename point_type<Linestring1>::type,
+            typename point_type<Linestring2>::type,
+            Strategy
+        >::type apply(Linestring1 const& ls1, Linestring2 const& ls2, Strategy const& strategy)
+    {
+        
+        typedef typename distance_result
+            <
+                typename point_type<Linestring1>::type,
+                typename point_type<Linestring2>::type,
+                Strategy
+            >::type result_type;
+        typedef typename boost::range_size<Linestring1>::type size_type;
+        
+        result_type dis_max=0,dis_min;
+        result_type const infinite = std::numeric_limits<double>::max();
```

Comment:
`infinite` is not needed right? Also using `std::numeric_limits<double>::max()` is wrong since user-define coordinate type can support values greater than that.

Reply:
I forgot to remove it. I have removed it in the updated code.